### PR TITLE
Add option to specify tenant for the to channel token

### PR DESCRIPTION
**Pull request by mingweiw**

### CODE REVIEW

**Comment by cleemullins** [comment]
@mingweiw, @Jeffders We don't want to merge this code until the actual fix (which is already published in C#) has been verified as working in the customer's environment. Once we know the solution works, we can merge this JS code in and consider pushing a new release. Until we are confident of the E2E solution working, we should hold off on this.

File: Node/core/src/bots/ChatConnector.ts [hunk 1]
```diff
@@ -150,6 +152,17 @@ export class ChatConnector implements IConnector, IBotStorage {
         };
     }
 
+    private getRefreshEndpoint(channelAuthTenant?: string): string {
+        var tenant = channelAuthTenant && channelAuthTenant.length > 0
+            ? channelAuthTenant
+            : "botframework.com";
```
Comment (cleemullins) [inline Node/core/src/bots/ChatConnector.ts]: Constant doesn't belong here.

File: Node/core/src/bots/ChatConnector.ts [hunk 2]
```diff
@@ -150,6 +152,17 @@ export class ChatConnector implements IConnector, IBotStorage {
...
+
+        var endpoint = 'https://login.microsoftonline.com/' + tenant + '/oauth2/v2.0/token';
```
Comment (cleemullins) [inline Node/core/src/bots/ChatConnector.ts]: constants should go in better locations.

File: Node/core/src/bots/ChatConnector.ts (hunk 1 above)
Comment (mingweiw) [reply Node/core/src/bots/ChatConnector.ts]: moved.

File: Node/core/src/bots/ChatConnector.ts (hunk 2 above)
Comment (mingweiw) [reply Node/core/src/bots/ChatConnector.ts]: yep

**Pull request merged**